application.py

```diff
@@ -135,43 +147,56 @@ def collapse_year(date):
 app.layout = layout
 
 
-@app.callback(Output("tally", "figure"), [Input("day_range", "value")])
-def update_tally(day_range):
+@app.callback(Output("tally", "figure"), [Input("area", "value")])
+def update_tally(area):
     """ Generate daily tally count """
 
-    today_doy = int(datetime.now().strftime("%j"))
-
     # Slice for April 1 (Day 92) - today.  Show 45 days minimum.
-    end_doy = today_doy if (today_doy - data_start_doy) >= 45 else data_start_doy + 45
-    de = df  # copy for local slicing
-    de = de.loc[
-        (de["doy"] >= data_start_doy) & (de["doy"] <= end_doy)
-    ]
+    # TODO -- lines below are commented out for intial work,
+    # but, we do need to be determining how this should work.
+    # today_doy = int(datetime.now().strftime("%j"))
+    # end_doy = today_doy if (today_doy - data_start_doy) >= 45 else data_start_doy + 45
+
+    # Spatial clip
+    if area == "ALL":
+        de = df.groupby(["FireSeason", "doy", "date_stacked"]).sum().reset_index()
+        print(de)
+    else:
+        de = df.loc[
+            (de["ProtectionUnit"] == area)
```

Comment:
Think this should be `df["ProtectionUnit"]` instead of `de`? Getting referenced before assignment error